testing/trino-testing/src/main/java/io/trino/testing/TestingTrinoClient.java

```diff
@@ -156,14 +156,15 @@ public void setStatementStats(StatementStats statementStats)
         @Override
         public void addResults(QueryStatusInfo statusInfo, QueryData data)
         {
-            if (types.get() == null && statusInfo.getColumns() != null) {
+            if ((types.get() == null || types.get().isEmpty()) && statusInfo.getColumns() != null) {
```

Comment:
Why this OR here?
This looks like we're overwriting `types` information after it's set for the first time.
I hope this is not needed, since it would be confusing to client tools.